src/shared/pretty-print.h

```diff
@@ -55,5 +55,21 @@ bool shall_tint_background(void);
 
 void draw_progress_bar(const char *prefix, double percentage);
 void clear_progress_bar(const char *prefix);
-void draw_progress_bar_unbuffered(const char *prefix, double percentage);
-void clear_progress_bar_unbuffered(const char *prefix);
+void draw_progress_bar_impl(const char *prefix, double percentage);
+void clear_progress_bar_impl(const char *prefix);
+
+static inline void flush_and_disable_bufferingp(FILE **p) {
+        assert(p);
+
+        if (*p) {
+                fflush(*p);
+                setvbuf(*p, NULL, _IONBF, 0); /* Disable buffering again */
```

Comment:
"again" seems a bit confusing here?

Reply:
Well the "again" was there because this was supposed to be used for stderr, which has buffering off by default - thus "again". It was off, then turned on for a while, then turned back off again.

Which I guess brings me to my slight concern - this might do unexpected things to streams that are already buffered. i.e. `WITH_BUFFERING(stdout)` will have the side-effect of turning off buffering on the stdout stream - which seems unexpected.

In practice, most everything (except for stderr) will have buffering on, no? `fopen` buffers by default, so the other stdio streams are buffered. So perhaps it's worth making this `WITH_BUFFERED_STDERR()` or so, and remove the choice of stream?

Reply:
@AdrianVovk Thank you for the suggestion. Renamed the macro, and dropped the choice of stream.